### DISCUSSION TITLE
Render frame by frame or at very low FPS?

### DISCUSSION THREAD

**Discussion by hisschemoller** [enable3d/enable3d#186] [Help / Questions] [answered]
Is it possible to manually advance frame by frame? So that I can step through the frames one by one and grab and save the canvas imageData? To collect an image sequence from which I can generate a video later with - for example - FFMPEG.

I used to do this by calling WebGLRenderer.render() in a three.js project, but I can't find an option with enable3d.

My 3D scene might be heavy for my laptop so I need to give each frame some time to render before I can grab it.

**Comment by yandeu** [comment, marked as answer]
`setAnimationLoop()` is set automatically when starting a scene.

You could overwrite it and call the private method `_update()` yourself.

See:

````ts
import { PhysicsLoader, Project, Scene3D } from 'enable3d'

class MainScene extends Scene3D {
  async create() {
    this.warpSpeed()

    this.physics.add.box({ y: 5 })

    // @ts-ignore
    setInterval(() => this._update(), 500) // run at 2 FPS
  }

  update() {
    this.renderer.setAnimationLoop(null)
  }
}

PhysicsLoader('/path', () => new Project({ scenes: [MainScene] }))
````

**Reply by hisschemoller** [reply]
Oh yes, that looks good. I'll try it tomorrow, too late now. Thanks very much!

**Reply by yandeu** [reply]
@hisschemoller I found your project a nice idea and coded it.

Check [this repository](https://github.com/enable3d/three-to-movie). It renders your scene into a .mp4 file. And it does this headlessly using puppeteer. 😍
